Week4 authorization implemented

Xin128 opened the pull request:
Week 4 authorizations features implemented except the optional ones

Xin128 committed:
just for testing userService info

Xin128 committed:
display message and login url working

Xin128 committed:
merge two files

Merge branch 'Week4Auth' of https://github.com/Xin128/Google_Intern_Projects into Week4Auth

Xin128 committed:
hide/display working

Xin128 committed:
authoriazation last two step finish

Xin128 committed:
merge conflicts and number of comments resolved

Xin128 committed:
first comparision

Xin128 committed:
charts 80% completed

Xin128 committed:
draft all done

Xin128 committed:
minor styling issues

Xin128 committed:
delete unused import statements

Xin128 committed:
updates
Merge branch 'Week4Auth' of https://github.com/Xin128/Google_Intern_Projects into Week4Charts

cisxingg commented on portfolio/src/main/webapp/script.js:
```diff
@@ -47,3 +44,25 @@ function deleteAllComments() {
+  fetch("/userLogIn")
+    .then((response) =>response.json())
+    .then((loginfo) => {
+      if (loginfo.status == 1) {
```
Maybe add a comment to explain what 1 is?

cisxingg commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -55,19 +58,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+
+    // Add limited number of comment contents and user email to the userComment Map  
+    HashMap<String, ArrayList<String>> userComment = new HashMap<String, ArrayList<String>>();
     for (Entity commentEntity:results.asList(FetchOptions.Builder.withLimit(maxNumComments))) {
```
Missed this last time: space in between "commentEntity", ":" and "results":
"commentEntity : results..."

cisxingg commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -55,19 +58,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+        String commentUser= (String)commentEntity.getProperty(USEREMAIL_PROPERTY);
+        ArrayList<String> msglist = userComment.get(commentUser);
+        if (msglist == null) {
+            msglist = new ArrayList<String>();
```
nit: spacing.

cisxingg commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -55,19 +58,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+        if (msglist == null) {
+            msglist = new ArrayList<String>();
+            msglist.add(commentMsg);
+            userComment.put(commentUser, msglist);
```
Suggestion: We can reduce the # of lines if you do:
userComment.put(commentUser, new ArrayList<String>(commentMsg));

Xin128 replied:
new ArrayList(commentMsg) is not working for Java 8. But I changed it by initializing in a line like following: new ArrayList<String>(Arrays.asList(commentMsg))

cisxingg commented on portfolio/src/main/java/com/google/sps/servlets/UserLogInServelet.java:
```diff
@@ -0,0 +1,60 @@
+  */
+@WebServlet("/userLogIn")
+public class UserLogInServelet extends HttpServlet {
+  HashMap<String,Object> userInfo = new HashMap<String, Object>();
```
nit: space in between "," and Object

cisxingg commented on portfolio/src/main/java/com/google/sps/servlets/UserLogInServelet.java:
```diff
@@ -0,0 +1,60 @@
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+      userInfo.put("status", 1);
+      userInfo.put("email", userEmail);
+      userInfo.put("logoutUrl", logoutUrl);
```
Instead of HashMap<String, Object>, do you think it is better to make an actual class UserInfo that has different fields?

Xin128 replied:
I was originally thinking of that, but then I realize that it takes a little bit longer to convert to json from a class object than a HashMap. Since there is only three elements here, is it still necessary? If so, I can encapsulate them into a new user class.

cisxingg replied:
@warfa-google Do you know if there is a better way? If not I am ok with this change :)

warfa-google commented on portfolio/src/main/java/com/google/sps/servlets/UserLogInServelet.java:
```diff
@@ -0,0 +1,60 @@
+  * Note: We first added the Gson library dependency to pom.xml.
+  */
+@WebServlet("/userLogIn")
+public class UserLogInServelet extends HttpServlet {
```
UserLoginServlet

Xin128 committed:
changed based on Cis's code review

Xin128 committed:
remove comments

Xin128 committed:
change order of variable definition

cisxingg commented on portfolio/src/main/java/com/google/sps/servlets/UserLogInServelet.java:
```diff
@@ -0,0 +1,60 @@
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
```
We don't really need this variable: urlToRedirectToAfterUserLogsOut given the value is small, or if you do, move it outside of the if statement, since it has the same value as urlToRedirectToAfterUserLogsIn.

cisxingg reviewed (approved):
LG in general, a couple more comments :)

warfa-google reviewed (approved):
LGTM

Xin128 committed:
editions after code review

Xin128 committed:
Merge pull request #11 from Xin128/Week4Charts

Week4 charts

Xin128 committed:
after second code review

Xin128 committed:
merge from Charts
Merge branch 'Week4Auth' of https://github.com/Xin128/Google_Intern_Projects into Week4Auth
